src/main/webapp/app/course/tutorial-groups/tutorial-groups-management/tutorial-groups/tutorial-groups-management/tutorial-groups-export-button.component/tutorial-groups-export-button.component.ts

```diff
@@ -0,0 +1,136 @@
+    ngUnsubscribe = new Subject<void>();
+
+    @ViewChild('exportDialog') exportDialogRef: TemplateRef<any>;
+
+    @Input() courseId: number;
+
+    @Output() exportFinished: EventEmitter<void> = new EventEmitter();
+
+    selectAll = false;
+
+    selectedFields: string[] = [];
+    availableFields = [
+        { value: 'ID', selected: false },
+        { value: 'Title', selected: false },
+        { value: 'Day of Week', selected: false },
+        { value: 'Start Time', selected: false },
+        { value: 'End Time', selected: false },
+        { value: 'Location', selected: false },
+        { value: 'Campus', selected: false },
+        { value: 'Language', selected: false },
+        { value: 'Additional Information', selected: false },
+        { value: 'Capacity', selected: false },
+        { value: 'Is Online', selected: false },
+        { value: 'Students', selected: false },
+    ];
+
+    constructor(
+        private modalService: NgbModal,
+        private tutorialGroupsService: TutorialGroupsService,
+        private alertService: AlertService,
+    ) {}
+
+    openExportDialog(event: MouseEvent) {
+        event.stopPropagation();
+        const modalRef: NgbModalRef = this.modalService.open(this.exportDialogRef, {
+            size: 'lg',
+            scrollable: false,
+            backdrop: 'static',
+            animation: false,
+        });
+
+        from(modalRef.result)
+            .pipe(
+                catchError(() => EMPTY),
+                takeUntil(this.ngUnsubscribe),
+            )
+            .subscribe(() => {
+                this.exportFinished.emit();
+            });
+    }
+
+    toggleSelectAll() {
+        this.selectAll = !this.selectAll;
+        this.availableFields.forEach((field) => (field.selected = this.selectAll));
+        this.updateSelectedFields();
+    }
+
+    onFieldSelectionChange(field: any) {
+        field.selected = !field.selected;
+        this.selectAll = this.areAllFieldsSelected();
+        this.updateSelectedFields();
+    }
+
+    updateSelectedFields() {
+        this.selectedFields = this.availableFields.filter((field) => field.selected).map((field) => field.value);
+    }
+
+    areAllFieldsSelected(): boolean {
+        return this.availableFields.every((field) => field.selected);
+    }
+
+    exportCSV(modal: NgbModalRef) {
+        this.tutorialGroupsService.exportTutorialGroupsToCSV(this.courseId, this.selectedFields).subscribe({
+            next: (blob: Blob) => {
+                const a = document.createElement('a');
+                const objectUrl = URL.createObjectURL(blob);
+                a.href = objectUrl;
+                a.download = 'tutorial-groups.csv';
+                a.click();
+                URL.revokeObjectURL(objectUrl);
+                this.resetSelections();
+                modal.close();
+            },
+            error: () => {
+                alert('Failed to export CSV');
```

Comment:
Why don't you use the alertService here? Is this alert also shown nicely on the top right?
If that is intended, please remove the alertService from the constructor, since it is unused.

Reply:
i changed it 👍